gapis/trace/android/adreno/profiling_data.go

```diff
@@ -90,6 +90,45 @@ func extractTraceHandles(ctx context.Context, replayHandles *[]int64, replayHand
 	}
 }
 
+func fixContextIds(contextIDs []int64) []int64 {
+	// This is a temporary work to workaround a QC bug
+	// that causes first deviceID to be zero after a
+	// renderpass change in the same queue submit.
+
+	zeroIndices := make([]int, 0)
+	deviceID := int64(0)
+	multipleDevice := false
+
+	for i, v := range contextIDs {
+		if v == 0 {
+			zeroIndices = append(zeroIndices, i)
+			continue
+		}
+
+		if deviceID != v {
+			if deviceID == 0 {
+				deviceID = v
+				continue
```

Comment:
I think you can remove this `continue`

Reply:
Yes that's true, I deleted the else instead.